### PR TITLE
Add K.resize_images backend op.

### PR DESCRIPTION
**Pull request by julienr**
This allows to take advantage of tensorflow’s resize_images operator in UpSampling2D. This was discussed in #1293 .

This also modifies `test_upsampling_2d` to test with different dim_ordering.

### CODE REVIEW

**Comment by fchollet** [comment]
For now, tests are failing.

**Comment by julienr** [comment]
Oops sorry, I forgot to checkin some changes.

File: keras/backend/theano_backend.py [hunk 1]
```diff
@@ -251,6 +251,25 @@ def repeat_elements(x, rep, axis):
     '''
     return T.repeat(x, rep, axis=axis)
 
+
+def resize_images(X, height, width, height_factor, width_factor, dim_ordering):
+    '''Resize the images contained in a 4D tensor of shape
+    - [batch, channels, height, width] (for 'th' dim_ordering)
+    - [batch, height, width, channels] (for 'tf' dim_ordering)
+    by a factor of (height_factor, width_factor)
+    '''
```
Comment (fchollet) [inline keras/backend/theano_backend.py]: It is looking like `height_factor` and `width_factor` can only be positive integers. This should be specified in the docstring. The API makes it sound like `*_factor` could be a float (e.g. 0.75 for an output image with 75% of the original height).

Comment (fchollet) [inline keras/backend/theano_backend.py]: I am also wondering whether it is necessary to specify the original height and width as part of the arguments (this information is part of the tensor X).

**Comment by julienr** [comment]
I fixed your two comments (get size from tensor and be explicit about scaling factors in the docstring).

The travis tests for `tf/python3.4` is failing on `tests/test_loss_weighting.py`with a strange error
```
E   EOFError: Compressed file ended before the end-of-stream marker was reached
```

I thinks it's some download corruption on the MNIST dataset on the travis server. The test is passing on my machine.

**Comment by fchollet** [comment]
Seems to have been a flaky test.

**Comment by fchollet** [comment]
Looks good to me.

**Comment by thomasantony** [comment]
Is there any reason why only integer scaling factors are allowed? Why not allow floats there and then truncate the resulting shapes to integers?

**Pull request merged**